spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -497,6 +487,96 @@ object Recurrent extends ContainerSerializable {
     new Recurrent[T](batchNormParams)
   }
 
+  /**
+   * set the cells' output and gradInput to recurrent's output and gradInput
+   * to decrease the copy expense.
+   * Copy src tensor to dst tensor along timeDime, default timeDime 2, batchDim 1
+   * @param src
+   * @param dst
+   */
+  private[bigdl] def copy[@specialized(Float, Double) T: ClassTag](
+    src: ArrayBuffer[Tensor[T]], dst: Tensor[T], offset: Int): Unit = {
+    val timeSize = dst.size(2)
+    var t = 1
+    while ((t + offset) <= timeSize) {
+      copyToIndex(src(t -1), dst, t + offset)
+      t += 1
+    }
+  }
+
+  /**
+   * select srcIndex subset of the 2-th dimension from src, and copy to dst
+   * @param src
+   * @param srcIndex the index of 2-th dimension from src
+   * @param dst
+   */
+  private[bigdl] def selectCopy[@specialized(Float, Double) T: ClassTag](
```

Comment:
define batchDim and timeDim as a constant, instead of using 1 or 2 directly.

Reply:
ok